src/pages/index.js

```diff
@@ -2,101 +2,232 @@ import Card from "../components/Card.js";
-
-const addCardModal = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
-addCardModal.setEventListeners();
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "41597df2-eb93-4b41-8bf1-f4a542a9ea54",
+    "Content-Type": "application/json"
+  }
+});
 
-const profileEditModal = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
-profileEditModal.setEventListeners();
+//Class Instances
+const previewImageModal = new PopupWithImage({ popupSelector: "#preview-image-modal" });
+const popupDelete = new PopupWithDelete({ popupSelector: "#delete-card-modal" });
+
+const userInfo = new UserInfo(".profile__title", ".profile__description", ".profile__image");
+
+let section;
+
+api.getInitialCards().then((initialCards) => {
+  section = new Section(
+    {
+      items: initialCards,
+      renderer: (data) => {
+        const cardElement = new Card(
+          data,
+          "#card-template",
+          handleImageClick,
+          handleDelete,
+          handleLikeCard
+        );
+        section.addItems(cardElement.getView());
+      },
+    },
+    ".cards__list"
+  );
+  section.renderItems();
+});
+api.fetchUserInfo().then((userData) => {
+  userInfo.setUserInfo({
+    name: userData.name,
+    about: userData.about,
+  });
+  userInfo.setAvatar({ avatar: userData.avatar });
+});
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
+    const addCardPopUp = new PopupWithForm(
+      { popupSelector: "#add-card-modal" },
+      handleAddCardFormSubmit
+    );
 
-section.renderItems();
+    const editProfilePopUp = new PopupWithForm(
+      { popupSelector: "#profile-edit-modal" },
+      handleProfileEditFormSubmit
+    );
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+    const editAvatarPopup = new PopupWithForm(
+      { popupSelector: "#avatar-edit-modal" },
+      handleAvatarSubmit
+    );
 
 /*Functions*/
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick).getView();
+  const card = new Card(data, "#card-template",
+    handleImageClick,
+    handleDelete,
+    handleLikeCard
+  ).getView();
   return card;
 }
 
-//Event Handlers
-function handleProfileEditSubmit(inputValues) {// {title: jacke , description: exploer}
-  profileEditValidator.resetValidation();
-  userInfo.setUserInfo({name: inputValues.title, description: inputValues.description});
-  profileEditModal.close();
-  profileEditValidator.disableButton();
+function handleAvatarSubmit(inputValues) {
+  editAvatarPopup.renderLoading(true);
+  api
+    .editAvatar({ avatar: inputValues.link })
+    .then((newAvatar) => {
+      userInfo.setAvatar(newAvatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarPopup.renderLoading(false);
+      editAvatarPopup.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.

Fix that here and in all the other places